packages/editor/src/store/reducer.js

```diff
@@ -515,6 +523,51 @@ export const editor = flow( [
 	} ),
 } );
 
+/**
+ * Reducer returning the initial edits state. With matching shape to that of
+ * `editor.edits`, the initial edits are those applied programmatically, are
+ * not considered in prmopting the user for unsaved changes, and are included
+ * in (and reset by) the next save payload.
+ *
+ * @param {Object} state  Current state.
+ * @param {Object} action Action object.
+ *
+ * @return {Object} Next state.
+ */
+export function initialEdits( state = INITIAL_EDITS_DEFAULTS, action ) {
+	switch ( action.type ) {
+		case 'SETUP_EDITOR':
+			if ( ! action.edits ) {
+				break;
+			}
+
+			return action.edits;
+
+		case 'SETUP_EDITOR_STATE':
+			if ( 'content' in state ) {
+				return omit( state, 'content' );
+			}
+
+			return state;
+
+		case 'UPDATE_POST':
```

Comment:
Not sure I understand why we need to address this here? my thinking is that initialEdits should be "overriden" by regular edits anyway when saving, why do we need to change the "initial edits" when we "edit" a post?

Reply:
The issue we need to avoid is having an initial edit resurface when an `edits` key is unset by `UPDATE_POST`.

An alternative here (perhaps better even) is to unset keys from `initialEdits` when an (assumed-user-initiated) `EDIT_POST` occurs (i.e. the initial edit can be discarded at the point where a user defines their own value).

Reply:
How does a user `unset` an edit? `editPost( { something: undefined } )`? Can this be considered as revert to initial value, which means use initialEdit if defined?

Reply:
It becomes unset when the edit matches the saved value of the post. So if there's an initial edit lingering, it will become (wrongly) assumed as an edit.

Reply:
> It becomes unset when the edit matches the saved value of the post

Should this be: "become unset if it matches the initial edit if defined, the saved value otherwise?

Isn't it a bug now?

Reply:
(I'm fine with the current implementation, just trying to push the reasoning to see if we can't find something interesting :) )

Reply:
So say we have

```js
const savedValue = "";

const initialEdit = "Demo";
```

and if I understand properly we're going to assume we don't have anything to save if `edits` are empty.

So if the user do `edit( { title: "" } )` We shouldn't unset the edit right? because the empty title needs to be saved.

Reply:
> So if the user do `edit( { title: "" } )` We shouldn't unset the edit right? because the empty title needs to be saved.

This is true, and a good reason not to do the proposed refactoring of applying the difference of keys from an edit in unsetting from `initialEdits`.

But, I think in the branch as-is, this works as expected:

Initially:

```js
post = { title: '' };
initialEdits = { title: 'demo' };
edits = {};
```

Edit title to "non-empty":

```js
post = { title: '' };
initialEdits = { title: 'demo' };
edits = { title: 'non-empty' };
```

Edit title back to empty:

```js
post = { title: '' };
initialEdits = { title: 'demo' };
edits = {}; // title omitted because matches post value
```

Save will include `{ title: 'demo' }`

The `UPDATE_POST` will cause:

```js
post = { title: 'demo' }
edits = {}
initialEdits = {}
```

Reply:
> Edit title back to empty:

In this current branch at this ⬆️ point, the `isDirty` is true which means we can trigger a save but when you change the `isDirty` check to the presence of `edits`, it will be `false` right?

Reply:
That's correct.

Reply:
But that's wrong right? Because we still need to save the fact that we removed the "demo" from title

Reply:
Have I said how much I loathe our consideration of initial edits? 😄 I'll have to think on this one.

Reply:
> Is it true / should it be true that the button is enabled / disabled by whether the post is dirty? Or purely that it's saveable?

I think if the user has an empty title and a content, it should be enabled. At least that's how it works now.

Reply:
It becomes more an issue when considering the demo post, which should be immediately saveable even if technically `editor.edits` would be empty (since it's all initial edits).

Reply:
One thing I noticed which is weird though in our current behavior is that I'd have expected the "save" button to be disabled when you open the demo page without any modification. And if the current behavior is the right one, that I'm completely confused :P

Reply:
(I wrote the last comment without reading your response :P )

Reply:
👍